### PR TITLE
[MB-1822] Fixed null pointer issue when closing subscribers.

### PR DESCRIPTION
**Pull request by abeykoon**
Fixing https://wso2.org/jira/browse/MB-1822.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java [hunk 1]
```diff
@@ -312,8 +312,6 @@ public void closeConnection(UUID channelID, String nodeID) throws AndesException
                         + " on non durable subscription close");
             }
         }
-
-        this.subscriberConnection = null;
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java]: Why is it wrong to set it to null?

Comment (abeykoon) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java]: CQEngine indexes subscriptions by connected node etc. It needs this to be not null at any time we query. 

when closing subscriptions

1. close connection -- > here is where above code made it null
2. remove from subscription registry (CQEngine access)
 
At step 2 it looks for the values. Thus keeping it until removal is not a problem as we remove the subscription everywhere  we close it.

Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java]: Can we have jira ticket id in title

Comment (abeykoon) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java]: Done. Updated the title

**Pull request merged**